apps/api/src/queries/photograph/photograph-queries-fetch-by-id.e2e.spec.ts

```diff
@@ -0,0 +1,538 @@
+                            testUserWithGroups: new CoscradUserWithGroups(
+                                dummyUser.clone({
+                                    roles: [CoscradUserRole.projectAdmin],
+                                }),
+                                []
+                            ),
+                        }
+                    ));
+                });
+
+                describe(`when there is a photograph with the given Id`, () => {
+                    describe(`when the photograph is published`, () => {
+                        beforeEach(async () => {
+                            // TODO: we need to check that contributors come through
+
+                            await photographQueryRepository.createMany([targetPhotographView]);
+                        });
+
+                        it(`should return the expected result`, async () => {
+                            const res = await request(app.getHttpServer()).get(
+                                buildDetailEndpoint(photographId)
+                            );
+
+                            expect(res.status).toBe(httpStatusCodes.ok);
+
+                            const photograph = res.body as IDetailQueryResult<IPhotographViewModel>;
+
+                            const { actions } = photograph;
+
+                            // admin should see commands
+                            expect(actions).not.toHaveLength(0);
+
+                            /**
+                             * TODO We should add a separate test that checks
+                             * that the correct actions come through in different
+                             * scenarios.
+                             */
+                            expect(actions).toMatchSnapshot();
+                        });
+                    });
+
+                    describe(`when the term is not published`, () => {
+                        beforeEach(async () => {
+                            await photographQueryRepository.createMany([
+                                clonePlainObjectWithOverrides(targetPhotographView, {
+                                    isPublished: false,
+                                }),
+                            ]);
+                        });
+
+                        it(`should return the expected result`, async () => {
+                            const res = await request(app.getHttpServer()).get(
+                                buildDetailEndpoint(photographId)
+                            );
+
+                            expect(res.status).toBe(httpStatusCodes.ok);
+
+                            // Commands should be visible to admin
+                            expect(res.body.actions).not.toEqual([]);
+                        });
+                    });
+
+                    describe(`when the term is not published but the user has explicit access`, () => {
+                        // This case is a bit unclear: does this project admin have access to
```

Comment:
I'd say it's ok to update the comment to read:
"Currently, a project admin has access to all resources. In the future, this may not be the case. Nonetheless, a project admin will always have access to a resource for which their name appears in the query ACL."

Reply:
Edited comment